resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -0,0 +1,235 @@
+$global:tzAutoUpdatePath = "HKLM:\SYSTEM\CurrentControlSet\Services\W32Time\Parameters"
+$global:SysTrayPath = "HKCU:\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced"
+
+#region Functions 
+function TryGetRegistryValue
+{
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$Key,
+
+        [Parameter(Mandatory = $true)]
+        [string]$Property
+    )    
+
+    if (Test-Path -Path $Key)
+    {
+        try
+        {
+            return (Get-ItemProperty -Path $Key -Name $Property -ErrorAction SilentlyContinue | Select-Object -ExpandProperty $Property)     
+        }
+        catch
+        {
+            Write-Verbose "Property `"$($Property)`" could not be found."
+        }
+    }
+    else
+    {
+        Write-Verbose "Registry key does not exist."
+    }
+}
+#endRegion Functions
+
+#region Enum
+enum TimeZoneAutomatically 
+{
+    NTP
+    NoSync
+}
+
+#region Classes
+<#
+.SYNOPSIS
+    DSC Resource to manage Windows Time settings.
+
+.DESCRIPTION
+    This `Time` DSC Resource allows you to manage the time zone, automatic time zone update, and system tray date/time visibility settings on a Windows machine.
+
+.PARAMETER TimeZone
+    The time zone to set on the machine. The value should be a valid time zone ID from the list of time zones (Get-TimeZone -ListAvailable).Id. The default value is the current time zone.
+
+.PARAMETER SetTimeZoneAutomatically
+    The method to use to set the time zone automatically. The value should be either `NTP` or `NoSync`. The default value is `NTP`.
+
+.PARAMETER ShowSystemTrayDateTime
+    Whether to show the date and time in the system tray. The value should be a boolean. The default value is `$true`.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -Name Time -Method Set -Property @{ TimeZone = "Pacific Standard Time"; SetTimeZoneAutomatically = "NTP"; ShowSystemTrayDateTime = $true }
+
+    This example sets the time zone to Pacific Standard Time, sets the time zone to be updated automatically using NTP, and shows the date and time in the system tray.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -Name Time -Method Get -Property {}
+
+    This example gets the current time settings on the machine.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -Name Time -Method Test -Property @{ TimeZone = "Pacific Standard Time"}
+
+    This example tests whether the time zone is set to Pacific Standard Time.
+#>
+[DscResource()]
+class Time 
+{
+    # Timezone values are taken from the list of timezones (Get-TimeZone -ListAvailable).Id
+    # TODO: Track issue 125 on PSDesiredStateConfiguration repository to add a ValidateSet for time zones
+    [DscProperty(Key)]
+    # [ValidateSet(
```

Comment:
Should probably be a validate script; Ref your linked issue - https://github.com/PowerShell/PSDesiredStateConfiguration/issues/125#issuecomment-2456396017

Reply:
The only thing that I can think of, is create a enum without the spaces, and add another helper function to translate it. Might be a bit overkill, so I'm curious what the PS team is going to mention.